File: Common/apigatewayResponse.js
export const ApiGatewayResponse  = class {
    setStatus (status) {
        this.status = status; 
        return this;
    }
    setOutput (output) {
        this.output = output;
        return this;
    }
    setError (error) {
        this.error = error;
        return this;
    }
    setModule(module) {
        this.module = module;
        return this;
    }
};

File: CreatureService/Entities/APIGatewayResponse.cs
namespace CreatureService.Entities;

public class APIGatewayResponse<T> {
    public StatusCodes Status { get; set; }
    public T Output { get; set; }
    public string? Error { get; set; }
    public string? Module { get; set; }
}


public enum StatusCodes {
    Success = 200,
    ServerError = 500,
    BadRequest = 400,
    NotFound = 404,    
}